opsdroid/connector/twitch/__init__.py

```diff
@@ -0,0 +1,576 @@
+import os
+import re
+import logging
+import asyncio
+import aiohttp
+import websockets
+import json
+
+from voluptuous import Required
+
+import opsdroid.connector.twitch.events as twitch_event
+
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message
+from opsdroid.const import TWITCH_API_ENDPOINT, TWITCH_WEBHOOK_ENDPOINT, DEFAULT_ROOT_PATH
+
+
+CONFIG_SCHEMA = {
+    Required("code"): str,
+    Required("client-id"): str,
+    Required("client-secret"): str,
+    Required("channel"): str,
+    "redirect": str,
+    "forward-url": str
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+TWITCH_JSON = os.path.join(DEFAULT_ROOT_PATH, 'twitch.json')
+
+
+def build_api_url(client_id, client_secret, authorization, redirect):
+    """Build api url to authenticate and refresh token."""
+    clientid = f"?client_id={client_id}"
+    secret = f"&client_secret={client_secret}"
+    grant = f"&grant_type={authorization}"
+    redirect = f"&redirect_uri={redirect}"
+
+    return f"{TWITCH_API_ENDPOINT}{clientid}{secret}{grant}{redirect}"
+
+async def get_user_id(channel, token, client_id):
+    """Calls twitch api to get broadcaster user id.
+    
+    A lot of webhooks expect you to pass your user id in order to get the 
+    notification when a user subscribes or folllows the broadcaster
+    channel.
+    
+    """
+    url = f"https://api.twitch.tv/helix/users?login={channel}"
+    async with aiohttp.ClientSession() as session:
+        response = await session.get(url, headers={
+            'Authorization': f"Bearer {token}",
+            'Client-ID': client_id
+        })
+        
+        if response.status != 200:
```

Comment:
I recommend changing this to >=400 as there are many valid reasons why you would get a non 200 error with a valid response body. A temporary redirect for example.

```suggestion
        if response.status >= 400:
```